### ISSUE TITLE
question about instruction-Tuning LLM

### ISSUE THREAD

**Issue by zhongjian-zhang** [HKUDS/GraphEdit#1]
Hi, thank you for your interesting research.
I have some questions: 
1. The abstracts of some paper are missing, do you have a separate process for the missing values mentioned above?
2. The text length of the title of the paper as well as the abstract is relatively long, and a sample of the constructed instructions may go well beyond the common 2048, so have you dealt with this situation? What is the maximum token length in the paper?
3. Could you provide further information on the resource requirements for tune LLM?
Looking forward to your response!

**Comment by LarFii** [comment]
Thank you for your interest in GraphEdit.
For the questions:
1. Because the abstracts missing are mainly in Cora, so we designed a special prompt for Cora to skip this situation. You can see the prompt in [Cora](https://github.com/HKUDS/GraphEdit?tab=readme-ov-file#3-instruction-template).
2. Considering that the text length of the title and abstract of the paper may be too long, only two nodes are processed at a time, so this situation will not occur in the vast majority of cases. In actual testing, the maximum text length is rarely greater than 2048, and we will truncate it.
3. For tune LLM, we used one NVIDIA GeForce RTX 3090 and trained a Lora model.

**Comment by zhongjian-zhang** [comment]
Thank you for your response.
For the question1, why do not you derectly remove this edge which have insufficient text information? Have you tried the removal method?
For the question2, when considering both node information(including title and abstract) for each edge in Cora, about 25% edge will exceed 2048 characters, is this okay for fine-tuning LLM?
Expected your further reply!

**Comment by LarFii** [comment]
For question 1, if we directly remove this edge, which has insufficient text information, there will be some nodes becoming islets, which are difficult for GNN to classify.
For question 2, to ensure efficiency, we keep the current token limit to ensure the scalability of GraphEdit. For situations where information is lost due to truncation on the Cora dataset, GraphEdit tends to make confident judgments.
 
The following table shows the confusion matrix on the Cora dataset, and it can be seen that even with information loss caused by truncation, GraphEdit can still play a significant denoising role.

| 2056 (TN) | 1246 (FP) |
|--|--|
| 1112 (FN) | 14417 (TP) |

**Comment by zhongjian-zhang** [comment]
> Thank you for your interest in GraphEdit. For the questions:
> 
> 1. Because the abstracts missing are mainly in Cora, so we designed a special prompt for Cora to skip this situation. You can see the prompt in [Cora](https://github.com/HKUDS/GraphEdit?tab=readme-ov-file#3-instruction-template).
> 2. Considering that the text length of the title and abstract of the paper may be too long, only two nodes are processed at a time, so this situation will not occur in the vast majority of cases. In actual testing, the maximum text length is rarely greater than 2048, and we will truncate it.
> 3. For tune LLM, we used one NVIDIA GeForce RTX 3090 and trained a Lora model.

Hello, for the text length over 2048, how is it truncated when tune the LLM?
Expected your further reply!

**Comment by LarFii** [comment]
In ./LLM/scripts/train_lora.sh, we set model_max_length to 2048, which makes the tokenizer truncate text exceeding 2048. The tokenizer initialization in lines 208-214 of ./LLM/graphedit/train/train_lora.py.
